### PR TITLE
new moderator tools for removing links

### PR DESCRIPTION
**Pull request by YishaiGlasner**
## Description
new moderator tools for removing links

## Code Changes

- new helper function remove_links_from_csv in link.py
- add option to use this function in links_upload_api, when the request has `'action': 'DELETE'`
- new component for removing links in modtools

### CODE REVIEW

**Review by yitzhakc — CHANGES_REQUESTED** [review]

File: static/js/ModeratorToolsPanel.jsx [hunk 1]
```diff
@@ -396,6 +401,69 @@ class UploadLinksFromCSV extends Component{
...
+        <div className="remove-links-csv">
+            <div className="dlSectionTitle">Remove links</div>
+            <form id="remove-links-form" onSubmit={handleSubmit}>
+                <label>
+                    Upload file:
+                    <input type="file" name="csv_file"  onChange={handleFileChange} />
+                    <br/>
+                    Choose a csv file with two columns. First row should include titles, and the others valid refs to delete.
+                    <br/>
+                    Please note that it should be the exact ref, so 'Genesis 1' is different than 'Genesis 1:1-31'
```
[the remark this answers is not in the document]
Comment (YishaiGlasner) [reply static/js/ModeratorToolsPanel.jsx:456]: the instruction are for very few people. i think that the requester feedback is better here

File: sefaria/helper/link.py [hunk 2]
```diff
@@ -552,7 +552,33 @@ def add_links_from_csv(file, linktype, generated_by, uid):
...
+    errors_writer = csv.DictWriter(output, fieldnames=['ref1', 'ref2'])
+    errors_writer.writeheader()
+    for row in reader:
+        refs = sorted(row.values())
+        try:
+            link = Link().load({'refs': refs})
+            tracker.delete(uid, Link, link._id)
+            success += 1
+        except Exception as e:
+            print(e, refs)
```
[the remark this answers is not in the document]
Comment (YishaiGlasner) [reply sefaria/helper/link.py:572]: it was for debugging and forgotten there

File: sefaria/views.py [hunk 3]
```diff
@@ -1363,14 +1363,19 @@ def links_upload_api(request):
...
+        func = remove_links_from_csv
+        args = (file, uid)
+    else:
+        linktype = request.POST.get("linkType")
+        generated_by = request.POST.get("projectName") + ' csv upload'
+        func = add_links_from_csv
+        args = (file, linktype, generated_by, uid)
     try:
-        res = add_links_from_csv(file, linktype, generated_by, uid)
+        return jsonResponse({"status": "ok", "data": func(*args)})
```
[the remark this answers is not in the document]
Comment (YishaiGlasner) [reply sefaria/views.py:1376]: i'm not sure what you mean here. if the try statement fails we return HttpResponseBadRequest. if it succeeds, it means the csv file is ok, so the general status is ok. now, the user wants to know how many links were handled successfully and which links failed and why, so we send this information to the client in a csv file.

[the remark this answers is not in the document]
Comment (yitzhakc) [reply sefaria/views.py:1376]: Ah, you're correct, I mis-read the code. Thanks for pointing that out!

**Pull request merged**